libskk/state.vala

```diff
@@ -27,6 +27,13 @@ namespace Skk {
         "〕", "}", "]", "?", ".", ",", "!"
     };
 
+    public enum CompletionOrder {
```

Comment:
The code looks correct and I'm OK with merging this as it is, though I wonder if the option could be implemented as an array of "completion sources", so we can avoid manual permutation and allow future extension for non-dictionary sources.

Reply:
Thank you for your valuable feedback.
Your suggestion to use an array of "completion sources" seems beneficial for future extensibility. I will carefully consider this idea.